eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version4/ChannelCodecs4.scala

```diff
@@ -661,11 +668,15 @@ private[channel] object ChannelCodecs4 {
 
     val DATA_WAIT_FOR_CHANNEL_READY_01_Codec: Codec[DATA_WAIT_FOR_CHANNEL_READY] = (
       ("commitments" | commitmentsCodecWithoutFirstRemoteCommitIndex) ::
-        ("shortIds" | shortids)).as[DATA_WAIT_FOR_CHANNEL_READY]
+        ("shortIds" | shortids)).as[ChannelTypes4.DATA_WAIT_FOR_CHANNEL_READY_0b].xmap(d => d.migrate(), d => ChannelTypes4.DATA_WAIT_FOR_CHANNEL_READY_0b.from(d))
```

Comment:
I think we should only provide a decoder (which will make an encoding attempt explicitly fail), and remove the `.from()` companion methods:

```suggestion
        ("shortIds" | shortids)).as[ChannelTypes4.DATA_WAIT_FOR_CHANNEL_READY_0b].map(d => d.migrate()).decodeOnly
```

Reply:
Yes, and we use this already at several places for the same purpose.